### ISSUE TITLE
Support custom property names for QTree component

### ISSUE THREAD

**Issue by TNSaturday** [quasarframework/quasar#6661]
In Quasar's tree component we have **label** as a key for node-key and ticked as an array of strings of **labels**. Hierarchical structure is defined by **children** prop. Can we use custom names for this? For example I want ot use **names** props instead of **labels** and **roles** instead of **children**. The other thing is that I want my **ticked** array to be an array not of **labels**, but of another property, **id** for example. How can I obtain such a behaviour?

**Comment by mabele98** [comment]
Working on adding a feature to customize name (ex. name probs instead of labels).  

Regarding the ticked array request... Is this example you're describing having both **labels** and **id** listed in each node, or is this an example where **id** is used instead of **labels**?

**Comment by TNSaturday** [comment]
> Working on adding a feature to customize name (ex. name probs instead of labels).
> 
> Regarding the ticked array request... Is this example you're describing having both **labels** and **id** listed in each node, or is this an example where **id** is used instead of **labels**?

**Comment by TNSaturday** [comment]
Sorry, accidentally hit close and comment.

> Is this example you're describing having both labels and id listed in each node, or is this an example where id is used instead of labels?
Just IDs would be a start, storing an objects would be great in a tree component. I'll show both examples for clarity reasons. So in Quasar's docs we have QSelect: [QSelect](https://quasar.dev/vue-components/select#Affecting-model).
The options as you can see is an array of onjects:
![image](https://user-images.githubusercontent.com/27682909/79069188-f72c2d80-7cd4-11ea-9e27-cb1839323178.png)
What I want is in my model to have a name and id props instead of label and value accordingly:
```javascript
<script>
export default {
  data () {
    return {
      model: null,
      options: [
        {
          name: 'Google',
          id: '9d6e96c9-99bb-4274-8e48-7e74c7abf208'
        },
        {
          name: 'Facebook',
          id: '91793cb0-91bf-4321-bb3b-40b5bb0eb5c2 '
        },
        // so on
      ]
    }
  }
}
</script>
```

And I want to show names to the user but to update my selected field with an id.

 That's about the select. Now, for a tree. I want show to user the name props inside a Tree (remember that a tree itself may or may not be INSIDE of a select component). But in the selected prop I want to store the whole object, that was selected. Let's KISS it and stand to single selection as for now. Now it's label (again, from Quasars's docs):
```javascript
<script>
export default {
  data () {
    return {
      simple: [
        {
          label: 'Satisfied customers',
          children: [
            {
              label: 'Good food',
              children: [
                { label: 'Quality ingredients' },
                { label: 'Good recipe' }
              ]
            },
            {
              label: 'Good service (disabled node)',
              disabled: true,
              children: [
                { label: 'Prompt attention' },
                { label: 'Professional waiter' }
              ]
            },
            {
              label: 'Pleasant surroundings',
              children: [
                { label: 'Happy atmosphere' },
                { label: 'Good table presentation' },
                { label: 'Pleasing decor' }
              ]
            }
          ]
        }
      ],
      selected: 'Pleasant surroundings',
      ticked: [ 'Quality ingredients', 'Good table presentation' ],
      expanded: [ 'Satisfied customers', 'Good service (disabled node)', 'Pleasant surroundings' ]
    }
  }
}
```

See? These variables store strings, I want selected to be something like:
```javascript
selected: [
    {
        name: 'IDM service',
        url: 'https://someapi.com/api/v3',
        descriptiob: 'Lorem ipsum',
    },
    // so on
];
```

Like this.

**Comment by pdanpdan** [comment]
Closing this to track it in https://github.com/quasarframework/quasar/issues/6959

**Comment by pdanpdan** [comment]
Key and label keys were already configurable - I added PR for children key.
About using an object as key - you can do it already and ignore the vue warning about keys (but I don't encourage it), but it's much better from a performance point of view to keep it to strings and convert to object when you use it.

**Comment by rstoenescu** [comment]
New prop "children-key" will be available in "quasar" v1.11.3.

**Comment by shomalikhashayar** [comment]
I need a property or slot for hide or change no nodes available.

` <template v-slot:noNodes>
    <div>No nodes available.</div>
  </template>`
